document-generator-accounts/src/main/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImpl.java

```diff
@@ -34,27 +38,32 @@ public DocumentInfo getDocumentInfo() {
             return null;
         }
 
-        return Optional.of(transaction)
+        String resourceLink =  Optional.of(transaction)
                 .map(Transaction::getResources)
                 .map(resources -> resources.get(resourceId))
                 .map(Resource::getLinks)
                 .map(links -> links.get("resource"))
-                // when abridged has been migrated to use the company-accounts api, the code for the
-                // company accounts should work for abridged, resulting in this abridged specific code
-                // qualifying for removal
-                .filter(this::isAbridged)
-                .map(accountsLinks -> new DocumentInfo())
-                .orElse(null);
+                .orElseGet(() -> {
+                    LOG.info("Unable to find resource in transaction: " + resourceId);
+                    return "";
+                });
+
+
+        // when the Accounts migration has been completed to Company Accounts, this code can be removed
+        if (isAccounts(resourceLink)) {
+            return accountsHandler.getAccountsData(resourceLink);
+         }
 
+        return null;
```

Comment:
I know this is a skeleton and you are trying to keep the PR small but errors should be handled appropriately. You are returning a mixture of null and "" returns in this class.

At this point an OperationUnsupported exception could be thrown to clearly show that other account types are not yet supported. This would make it clear where the code should be extended to support other account types.

The other error returns in this class are all caused by the user supplying a URL to missing/incomplete data, I believe from a previous conversation with the architects that this will cause a 400 error. The return type/exception should be consistent and I believe should be in this PR.

Reply:
As mentioned in the PR description the exception handling hasn't been design or discussed yet so I will leave it until that has happened